components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -160,6 +160,25 @@ interface Engine {
         UnsupportedOperationException("Web extension support is not available in this engine")
     )
 
+    /**
+     * Invoked when a web extension has changed its permissions while trying to update to a
+     * new version. This requires user interaction as the updated extension will not be installed,
+     * until the user grants the new permissions.
+     *
+     * @param current The current [WebExtension].
+     * @param updated The update [WebExtension] that requires extra permissions.
+     * @param newPermissions Contains a list of all the new permissions.
+     * @param onPermissionsGranted A callback to indicate if the new permissions from the [updated] extension
+     * are granted or not.
+     */
+    fun onUpdatePermissionRequest(
```

Comment:
See comment above why we don't need to add this to the engine.

Reply:
Fixed!